elastic/kibana
# [Infrastructure UI] Hosts View: Filtering by metadata functionality

Closes #151010

## Summary

This PR adds filtering functionality to the metadata table. After some discussions there are changes in the design - see the [last comments](https://github.com/elastic/kibana/issues/151010#issuecomment-1513005095)

## Additional changes

As the filters will change the order I changed the id to not depend on the table index anymore (used os instead) inside [hosts table
](https://github.com/elastic/kibana/pull/155170/files#diff-168ba138bc6696100078e3e9cbc438ed7646d6336e9ef85a9c88553c9d8956f5)

## Testing
- Open the single host view for one of the hosts available in the table
- Inside the flyout select the Metadata tab
- Click on any of the available filter icons to add a filter 
- Check if the filter is applied and a toast is displayed
- Click on the same filter icon to remove it and check that it is removed
- If there are other filters applied they should combined with the metadata filters (so if filter A is added and then a metadata filter is applied both filters should be visible)


https://user-images.githubusercontent.com/14139027/232837814-7292c7ec-8b63-4172-bcd9-2d49daf3b145.mov
